hibernate-core/src/main/java/org/hibernate/id/enhanced/SequenceStyleGenerator.java

```diff
@@ -280,6 +286,13 @@ private int adjustIncrementSize(
 		return determineAdjustedIncrementSize( optimizationStrategy, incrementSize );
 	}
 
+	private boolean isSchemaToBeRecreated(String contributor, ConfigurationService configurationService) {
+		Set<ActionGrouping> actions = ActionGrouping.interpret( Collections.singleton(contributor), configurationService.getSettings() );
+		return actions.stream().anyMatch(
+				actionGrouping -> actionGrouping.getDatabaseAction() == Action.CREATE || actionGrouping.getDatabaseAction() == Action.CREATE_DROP
+		);
```

Comment:
```suggestion
		final Set<ActionGrouping> actions = ActionGrouping.interpret( Collections.singleton( contributor ), configurationService.getSettings() );
		final Action databaseAction = groupings.iterator().next().getDatabaseAction();
		return databaseAction == Action.CREATE || databaseAction == Action.CREATE_DROP;
```

Reply:
You can avoid using `stream()` since we know the `ActionGrouping` set will always contain a single element.

Reply:
> You can avoid using `stream()` since we know the `ActionGrouping` set will always contain a single element.

Ah, indeed, good catch, I missed that one.